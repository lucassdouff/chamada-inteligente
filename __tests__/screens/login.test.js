import React from 'react';
import renderer from 'react-test-renderer';
import LoginScreen from '../../src/app/screens/login/Login';
import { render, fireEvent } from '@testing-library/react-native';

const mockedDispatch = jest.fn();

jest.mock("@react-navigation/native", () => {
  const actualNav = jest.requireActual("@react-navigation/native");
  return {
    ...actualNav,
    useNavigation: () => ({
      navigate: jest.fn(),
      dispatch: mockedDispatch,
    }),
  };
});

describe('Login screen should render correctly and be interactable', () => {

    beforeEach(() => {
        mockedDispatch.mockClear();
    });

    it('renders correctly', () => {
        const loginScreen = renderer.create(<LoginScreen />).toJSON();
        expect(loginScreen).toMatchSnapshot();
    });

    it('should apply the value when changing text', () => {
        const { getByTestId } = render(<LoginScreen />);
        
        const enrollmentInput = getByTestId('enrollment-input');
        fireEvent.changeText(enrollmentInput, '220031133');
        
        const passwordInput = getByTestId('password-input');
        fireEvent.changeText(passwordInput, '12346');
        
        const foundButton = getByTestId('login-button');
        
        expect(enrollmentInput.props.value).toBe('220031133');
        expect(passwordInput.props.value).toBe('12346');
        expect(foundButton).toBeTruthy();
    });
    
});
