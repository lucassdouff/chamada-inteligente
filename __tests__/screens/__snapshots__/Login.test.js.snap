// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Login screen should render correctly and be interactable renders correctly 1`] = `
<RCTScrollView
  contentContainerStyle={
    {
      "flexGrow": 1,
      "justifyContent": "center",
    }
  }
  style={
    [
      {
        "width": "100%",
      },
      {
        "flexBasis": "0%",
        "flexGrow": 1,
        "flexShrink": 1,
      },
      {
        "paddingBottom": 16,
        "paddingLeft": 16,
        "paddingRight": 16,
        "paddingTop": 16,
      },
    ]
  }
>
  <View>
    <Image
      source={1}
      style={
        [
          {
            "alignSelf": "center",
          },
          {
            "marginBottom": 40,
          },
          {
            "height": 150,
            "width": 140,
          },
        ]
      }
    />
    <Text
      style={
        [
          {
            "alignSelf": "center",
          },
          {
            "marginBottom": 40,
          },
          {
            "fontSize": 20,
            "lineHeight": 28,
          },
        ]
      }
    >
      Acesso aos sistemas da UFF
    </Text>
    <Text
      style={
        [
          {
            "marginBottom": 8,
          },
        ]
      }
    >
      Identificação (idUFF)
    </Text>
    <TextInput
      collapsable={false}
      handlerTag={1}
      handlerType="NativeViewGestureHandler"
      onChangeText={[Function]}
      onGestureHandlerEvent={[Function]}
      onGestureHandlerStateChange={[Function]}
      placeholder="CPF, email, passaporte"
      style={
        [
          {
            "paddingLeft": 8,
            "paddingRight": 8,
          },
          {
            "paddingBottom": 2,
            "paddingTop": 2,
          },
          {
            "borderBottomWidth": 1,
            "borderLeftWidth": 1,
            "borderRightWidth": 1,
            "borderTopWidth": 1,
          },
          {
            "borderBottomColor": "#9ca3af",
            "borderLeftColor": "#9ca3af",
            "borderRightColor": "#9ca3af",
            "borderTopColor": "#9ca3af",
          },
          {
            "borderBottomLeftRadius": 4,
            "borderBottomRightRadius": 4,
            "borderTopLeftRadius": 4,
            "borderTopRightRadius": 4,
          },
          {
            "marginBottom": 16,
          },
        ]
      }
      testID="enrollment-input"
      underlineColorAndroid="transparent"
      value=""
    />
    <Text
      style={
        [
          {
            "marginBottom": 8,
          },
        ]
      }
    >
      Senha
    </Text>
    <TextInput
      collapsable={false}
      handlerTag={2}
      handlerType="NativeViewGestureHandler"
      onChangeText={[Function]}
      onGestureHandlerEvent={[Function]}
      onGestureHandlerStateChange={[Function]}
      style={
        [
          {
            "paddingLeft": 8,
            "paddingRight": 8,
          },
          {
            "paddingBottom": 2,
            "paddingTop": 2,
          },
          {
            "borderBottomWidth": 1,
            "borderLeftWidth": 1,
            "borderRightWidth": 1,
            "borderTopWidth": 1,
          },
          {
            "borderBottomColor": "#9ca3af",
            "borderLeftColor": "#9ca3af",
            "borderRightColor": "#9ca3af",
            "borderTopColor": "#9ca3af",
          },
          {
            "borderBottomLeftRadius": 4,
            "borderBottomRightRadius": 4,
            "borderTopLeftRadius": 4,
            "borderTopRightRadius": 4,
          },
          {
            "marginBottom": 16,
          },
        ]
      }
      testID="password-input"
      underlineColorAndroid="transparent"
      value=""
    />
    <View
      accessibilityRole="button"
      accessibilityState={
        {
          "busy": undefined,
          "checked": undefined,
          "disabled": undefined,
          "expanded": undefined,
          "selected": undefined,
        }
      }
      accessibilityValue={
        {
          "max": undefined,
          "min": undefined,
          "now": undefined,
          "text": undefined,
        }
      }
      accessible={true}
      collapsable={false}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        {
          "opacity": 1,
        }
      }
      testID="login-button"
    >
      <View
        style={
          [
            {},
          ]
        }
      >
        <Text
          style={
            [
              {
                "color": "#007AFF",
                "fontSize": 18,
                "margin": 8,
                "textAlign": "center",
              },
            ]
          }
        >
          ACESSAR
        </Text>
      </View>
    </View>
  </View>
</RCTScrollView>
`;
